apps/blockchain/lib/blockchain/block_header_info.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Blockchain.BlockHeaderInfo do
+  @moduledoc """
+  Module to interact with the block header in the blockchain.
+  """
+
+  @behaviour EVM.BlockHeaderInfo
+
+  @type t :: %__MODULE__{
+          block_header: Block.Header.t(),
+          db: MerklePatriciaTree.DB.db()
+        }
+
+  defstruct [:block_header, :db]
+
+  @doc """
+  Returns a new block interface.
+
+  ## Examples
+
+      iex> block_header = %Block.Header{}
+      iex> db = MerklePatriciaTree.Test.random_ets_db(:new_block_header_info)
+      iex> Blockchain.BlockHeaderInfo.new(block_header, db)
+      %Blockchain.BlockHeaderInfo{
+        block_header: %Block.Header{},
+        db: {MerklePatriciaTree.DB.ETS, :new_block_header_info}
+      }
+  """
+  def new(block_header, db) do
+    %__MODULE__{
+      block_header: block_header,
+      db: db
+    }
+  end
+
+  @doc """
+  Returns the header of the currently being processed block.
+
+  ## Examples
+
+      iex> %Block.Header{number: 10}
+      ...> |> Blockchain.BlockHeaderInfo.new(nil)
+      ...> |> Blockchain.BlockHeaderInfo.get_block_header()
+      %Block.Header{number: 10}
+  """
+  @impl true
+  def get_block_header(block_header_info) do
+    block_header_info.block_header
+  end
+
+  @doc """
+  Returns a block that is `n` blocks before the current block.
+
+  ## Examples
+
+      iex> db = MerklePatriciaTree.Test.random_ets_db()
```

Comment:
What do you think about moving this "example" to a test file?

Reply:
@ayrat555 we already have a very similar test in the test file. Would you prefer removing this example altogether or should we keep it?

Reply:
yes, I think we should get rid of these large doctests

Reply:
Sounds good to me! Updated in e0efead483f9cf729d1da10376a41b654508951a